src/settings/components/ExternalDataSourcesConfig/ExternalDataSourcesEdit.js

```diff
@@ -107,27 +116,48 @@ export default class ExternalDataSourcesEdit extends React.Component {
               dataOptions={
                 [{ value: '1', label: 'Package' }]
               }
+              disabled={isReadonly}
               label={<FormattedMessage id="ui-local-kb-admin.settings.externalDataSources.recordType" />}
               name={`${name}.rectype`}
-              required
+              required={!isReadonly}
               validate={required}
             />
           </Col>
         </Row>
-        <Field
-          component={TextField}
-          data-test-external-data-source-uri
-          label={<FormattedMessage id="ui-local-kb-admin.settings.externalDataSources.uri" />}
-          name={`${name}.uri`}
-          required
-          validate={v => { return (v && v.length) ? validateURLIsValid(v) : required(v); }}
-        />
+        <Row>
+          <Col xs={8}>
+            <Field
+              component={TextField}
+              data-test-external-data-source-uri
+              disabled={isReadonly}
+              label={<FormattedMessage id="ui-local-kb-admin.settings.externalDataSources.uri" />}
+              name={`${name}.uri`}
+              required={!isReadonly}
+              validate={v => {
+                if (!isReadonly) {
+                  return (v && v.length) ? validateURLIsValid(v) : required(v);
```

Comment:
`validateURLIsValid` can handle empty strings so you shouldn't need to check string length. It does look like you want to run both validators, which you can do with [`composeValidators`](https://github.com/folio-org/ui-erm/blob/92da603c4d70d7928bdb109072d7ea68f066d72f/src/components/AgreementLinesFieldArray/CustomCoverageField.js#L61)

Reply:
I've actually not really changed this one I don't think, it was previously:
`validate={v => { return (v && v.length) ? validateURLIsValid(v) : required(v); }}`
. When the field is empty, we want a marker saying "Hey, this should have something in it", and when it is _not_ empty we want to check it's a valid URL.

Reply:
No point saying an empty string isn't a valid URL right? That should be obvious

Reply:
Ahh, ok, didn't know this was there before. But `composeValidators` is useful because it does step through the validators one by one. So by ordering the validator functions, it would first check if it's filled in via one validator (reporting an appropriate message if it's not), and _then_ check if it's valid via the URL validator.